src/main/java/bookstore/exeptions/CustomGlobalExceptionHandler.java

```diff
@@ -0,0 +1,48 @@
+package bookstore.exeptions;
+
+import java.time.LocalDateTime;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request) {
+
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> mappedErrors = ex.getBindingResult()
+                .getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", mappedErrors);
```

Comment:
Currently you only return the message strings. It’s often helpful to include the field path too, e.g.:

> { "field": "price", "error": "must be positive" }


```suggestion
                List<Map<String, String>> errors = ex.getBindingResult()
            .getFieldErrors()
            .stream()
            .map(error -> Map.of(
                "field", error.getField(),
                "error", error.getDefaultMessage()))
            .collect(Collectors.toList());

        body.put("errors", errors);
```